fortify-scan.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/sh
+alias echo='echo $(date +"%F %T")'
+now=$(date +"%s")
+projectName=$(basename -- "$(pwd)")
+scanFolder="fortify-scans/$projectName"
+logfile="$scanFolder/sourceanalyzer.log"
+
+if [ ! -d "$scanFolder" ]; then
+  git clone git@github.com:lightbend/fortify-scans.git
+  if [ ! -d "$scanFolder" ]; then
+    mkdir "$scanFolder"
+  fi
+fi
+
+echo Starting analysis of scans
+for D in ~/.fortify/sca21.1/build/*; do
+  if [ -d "${D}" ]; then
+    d="$(basename -- "$D")"
+    echo Starting "$d" >> "$logfile"
+    sourceanalyzer -b "$d" -f "$scanFolder/$d".fpr -scan
+    echo Finished "$d" >> "$logfile"
+    echo Finished "$d"
+  fi
+done
+echo Scans complete
+
+for F in "$scanFolder"/*.fpr; do
+  f="$(basename -- "${F%.*}")"
+  echo Generating report for "$f"
+  BIRTReportGenerator -template "Developer Workbook" -source "$F" -format PDF -output "$scanFolder/$f.pdf"
+done
+
+echo Pushing scan results to repo
+cd fortify-scans || exit
+git add .
+commitMsg="Scan performed on $(date +"%Y-%m-%d_%T")"
+git commit -m "$commitMsg"
+git push origin
```

Comment:
Should this really push to the repo? What is the process for running these scripts? Are they supposed to be run by CI or is it manually from a local machine?

Reply:
Right now, it's manual but I am writing this with the idea of automation. Right now, it's manual but I am writing this with the idea of automation, so CI is the goal. 

As for pushing to the repo, I've changed the script to push the results to a private repo, lightbend/fortify-scans under a folder named for the project being scanned. Technically, anyone could have Fortify SCA and run the same scans but I figured it best to not expose our vulnerabilities so directly as to include with the project.